src/gobupload/relations.py

```diff
@@ -0,0 +1,149 @@
+'''
+Build relations for a specific collection from a catalog.
+
+We retrieve all relations for each source and create the relation on the
+_id field.
+
+'''
+from sqlalchemy import cast
+from sqlalchemy.types import Text
+
+from gobcore.model import GOBModel
+from gobcore.sources import GOBSources
+
+from gobupload.models.model import models
+from gobupload.storage.handler import GOBStorageHandler
+
+model = GOBModel()
+sources = GOBSources()
+storage = None
+
+
+def build_relations(catalog_name, collection_name):
+    """Build all relations for a collection from a catalog.
+
+    :param catalog_name: the name of the catalog
+    :param collection_name: the name of the collection
+    :return:
+    """
+    global storage
+
+    storage = GOBStorageHandler()
+
+    relations = sources.get_relations(catalog_name, collection_name)
+    for relation in relations:
+        _update_relation(catalog_name, collection_name, relation)
+
+
+def _update_relation(catalog_name, collection_name, relation):
+    """Update a specific relation for a collection from a catalog.
+
+    :param catalog_name: the name of the catalog
+    :param collection_name: the name of the collection
+    :param relation: the relation dict, containing all details about the relation
+    :return:
+    """
+    with storage.get_session():
+        # Get all objects we need to update
+        update_model = models[model.get_table_name(relation['catalog'], relation['collection'])]
+
+        # Get all entities which can be updated
+        entities = _get_all_entities(update_model, relation)
+        total_count = entities.count()
+        print(f"About to build {total_count} relation(s) for "
+              f"{relation['field_name']} on {relation['catalog']}_{relation['collection']} "
+              f"to {relation['destination_attribute']} on {catalog_name}_{collection_name}")
+
+        query = _get_update_query(catalog_name, collection_name, relation)
+        if query:
+            result = storage.session.execute(query)
+
```

Comment:
The following messages should be outside the with statement
The query is only commited at the end of the with statement